### PR TITLE
GCF: Add Pub/Sub publisher sample

### PR DESCRIPTION
**Pull request by ace-n**
**Partially** fixes the linked bug.

### CODE REVIEW

File: functions/pubsub/main.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# limitations under the License.
+
+# [START functions_pubsub_setup]
+import base64
+from google.cloud import pubsub_v1
+import json
+import os
+
+
+# Instantiates a client
```
Comment (grant) [inline functions/pubsub/main.py:22]: a Pub/Sub client

File: functions/pubsub/main.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+publisher = pubsub_v1.PublisherClient()
+PROJECT_ID = os.getenv('GCP_PROJECT')
+# [END functions_pubsub_setup]
+
+
+# [START functions_pubsub_publish]
+# Publishes a message to a Cloud Pub/Sub Topic.
```
Comment (grant) [inline functions/pubsub/main.py:29]: nit: Cloud Pub/Sub **t**opic. The word topic is not part of the product title.

**Comment by grant** [comment]
> **Partially** fixes the linked bug.

Please clearly link the bug. I can't find it.

**Comment by grant** [comment]
Oh, I think it's in the branch name. http://b/151080185

File: functions/pubsub/main.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    })
+    message_bytes = message_json.encode('utf-8')
+    print(message_bytes)
+
+    # Publishes a message
+    try:
+        publish_future = publisher.publish(topic_path, data=message_bytes)
+        publish_future.result()  # Verify the publish succeeded
+        return 'Message published.'
+    except Exception as e:
```
Comment (busunkim96) [inline functions/pubsub/main.py:55]: Is there a more specific exception that could be captured here?

Comment (ace-n) [reply functions/pubsub/main.py:55]: Not sure - though I think the intent of this was to "fail gracefully" as documented [here](https://cloud.google.com/functions/docs/bestpractices/tips#error_reporting)

**Pull request merged**